src/doc/src/reference/environment-variables.md

```diff
@@ -231,7 +231,7 @@ corresponding environment variable is set to the empty string, `""`.
 * `CARGO_PKG_RUST_VERSION` — The Rust version from the manifest of your package.
   Note that this is the minimum Rust version supported by the package, not the
   current Rust version.
-* `CARGO_CRATE_NAME` — The name of the crate that is currently being compiled.
+* `CARGO_CRATE_NAME` — The name of the crate that is currently being compiled, e.g the name of the library or the binary.
```

Comment:
To avoid further confusion, I might suggest saying it is the name of the [Cargo Target](https://doc.rust-lang.org/cargo/reference/cargo-targets.html) with `-` converted to `_`. And then "such as the name of the library, binary, example, integration test, or benchmark".

Reply:
The `CARGO_CRATE_NAME` can be modified by setting `[lib.name]`.